test/Microsoft.AspNetCore.Rewrite.Tests/MiddlewareTests.cs

```diff
@@ -65,6 +65,7 @@ public class MiddlewareTests
             var response = await server.CreateClient().GetAsync(new Uri("http://example.com"));
 
             Assert.Equal("https://example.com/", response.Headers.Location.OriginalString);
+            Assert.Equal(StatusCodes.Status301MovedPermanently, (int)response.StatusCode);
```

Comment:
Theory with different status codes.
Can you also add some variants for the other AddRedirectToHttps overloads?